### ISSUE TITLE
Don't allow manual additions/changes to use unqualified releases

### ISSUE THREAD

**Issue by nesl247** [dependabot/feedback#100]
I'm not sure if this is something Dependabot would ever want to do, but the name itself doesn't imply it couldn't be done.

What I am thinking is to have Dependabot check all PRs for changes to a package manager's config file, like `composer.json`, and stop a PR from being merged if it uses unqualified releases like `dev-master`.

It's highly unrecommended to use versions like that, and unless it's a specific dependency, like security update checkers, it should be avoided.

**Comment by feelepxyz** [comment]
@nesl247 yes interesting idea, think this would have to be implemented as a CI step that checks all commits and makes sure the dependency versions are legit releases. We don't have any immediate plans on adding this, but will keep this open as a reminder. Keen to figure out if there are more commit level checks we could bundle together in a CI step. Any ideas welcome!

**Comment by nesl247** [comment]
If you need more ideas for CI steps:

* Check if file is valid - the files usually work even though they don't pass the package manager's validator. For example, composer has `composer validate` but even if you are missing `description`, it doesn't stop it from working.

* We update dependencies manually sometimes, because it requires complicated updates, such as php-cs-fixer requiring new rules, or deprecations being added which causes unit test failures. Other times are updating from minor to major (we enable the minor updates only feature). Having Dependabot check to make sure there are no security issues introduced in the PR would be beneficial.

FYI @feelepxyz I'm guessing you work for dependabot, but oddly it doesn't show you as a member of the organization.

Thanks!

**Comment by greysteil** [comment]
He does! Sort that out Phil! 😉

**Comment by feelepxyz** [comment]
@nesl247 nice catch, fixed my org visibility 
Also great suggestions, there's definitely something we can do here.
